### DISCUSSION TITLE
Using num_samples = a value in the forecast code. Is there a way to know the indices of the random selection of the samples, and use the same exact samples' indices for another forecast? I don't want to reproduce the same results using random seeds. This is a different question.

### DISCUSSION THREAD

**Discussion by Msaleh87** [awslabs/gluonts#2698] [Q&A] [answered]
I have this simple code:

```
T = 0
test_data = ListDataset(
    [
        {"start": df_input.index[T], "target": df_input.Load[T:Threshold + prediction_horizon], 
         'feat_dynamic_real': [time_feat_df_5min_ready.DofW[T:Threshold + prediction_horizon]], 
        'time_feat': [time_feat_df_5min_ready.DayofWeek[T:Threshold + prediction_horizon],
                     time_feat_df_5min_ready.Hour[T:Threshold + prediction_horizon],]},     
    ],

freq = freq)

forecast_it, ts_it = make_evaluation_predictions(
    dataset = test_data,
    predictor = predictor,
    num_samples = 20)

forecasts = list(forecast_it)
tss = list(ts_it)
```

Now, how to know which 20 samples the model picked? What are their indices and how to use the same indices to forecast another time?
I don't want to reproduce the same results using random seeds. This is a different question.
Thanks

**Comment by mbohlkeschneider** [comment]
Hi @Msaleh87,

Thank you for your question. I'm not sure I understand. What does an "index" that you talking about refer to?

**Comment by Msaleh87** [comment]
Hi @mbohlkeschneider,

Let's say in the code above, I set num_samples = 3, then my understanding is that the forecast/ make_evaluation_predictions function will choose random 3 probabilistic paths for the future forecast and outcome the mean of them. Let's say the function picked sample paths 3,4,5 out of 100 possible future forecasts. 
Those 3,4,5 are the sample indices, each containing a possible forecast. So, in another forecast, if I want to pick the same samples' selection, how to do so?
Thank u!

**Comment by mbohlkeschneider** [comment, marked as answer]
Hi @Msaleh87, 

I think there might be a misunderstanding how sampling works. If you invoke `predict` (which makes evaluation predictions) does under the hood, it will return a tensor where one dimension will be the sample dimension. If you have `num_samples=3`, the size of this dimension is 3 (and the indices 0, 1, 2 accordingly). There is no mechanism of picking samples out of some larger set of samples. If you invoke `predict` again the model will generate another set of samples. 

Hope this was clear.

**Comment by Msaleh87** [comment]
Hi @mbohlkeschneider,

Thanks so much for this clarification, it makes sense now, u saved me a lot of time! So I guess my question is: Is there a way to generate the same set of samples? 
I guess the answer would be to use random.seed() or increase num_samples to a high value like 100. Is there any other way?

Also, let's say I generate num_samples = 100. Can I access and select 0,1,2 samples, and how?

Thanks so much for ur time and help

**Comment by mbohlkeschneider** [comment]
Hi @Msaleh87,

This is the object that the models with sampling return: https://github.com/awslabs/gluonts/blob/b0fb27f419e52f3b98e1b1bb6a078b5c7d57da14/src/gluonts/model/forecast.py#L423 First axis of the `.samples` attribute are the samples.
